src/icalendar/tests/test_issue_867_todo_duration_fix.py

```diff
@@ -0,0 +1,178 @@
+        ("P1W", timedelta(weeks=1)),
+        (
+            "P1Y2M3DT4H5M6S",
+            timedelta(days=428, hours=4, minutes=5, seconds=6),
+        ),  # Approx 1 year 2 months
+    ]
+
+    for duration_str, expected_delta in test_cases:
+        todo = Todo()
+        todo.add("UID", f"test-{duration_str}")
+        todo.add("DURATION", expected_delta)
+
+        assert todo.duration == expected_delta
+
+
+def test_todo_duration_maintains_backward_compatibility():
+    """Test that the fix doesn't break existing functionality."""
+    # Create todo the old way (DTSTART + DUE)
+    todo = Todo()
+    todo.add("UID", "test-backward-compat")
+    todo.add("SUMMARY", "Backward compatibility test")
+    todo.start = datetime(2026, 3, 19, 9, 0)
+    todo.end = datetime(2026, 3, 19, 17, 0)
+
+    # Should still work as before
+    assert todo.duration == timedelta(hours=8)
+
+    # Properties should still work
+    assert todo.start == datetime(2026, 3, 19, 9, 0)
+    assert todo.end == datetime(2026, 3, 19, 17, 0)
+
+
+def test_todo_duration_edge_case_only_dtstart():
+    """Test Todo with only `DTSTART` (no `DUE` or `DURATION`)."""
+    todo = Todo()
+    todo.add("UID", "test-only-start")
+    todo.start = datetime(2026, 3, 19, 12, 0)
+
+    # This should use the fallback logic from the end property
+    # which returns start + 1 day for date, or just start for datetime
+    assert todo.duration == timedelta(0)  # end defaults to start for datetime
+
+
+def test_issue_867_exact_reproduction():
+    """Test the exact scenario reported in issue #867."""
+    # This reproduces the caldav project test case that was failing
+    my_task = Todo.from_ical("""BEGIN:VTODO
+UID:taskwithoutdtstart
+DTSTAMP:20070313T123432Z
+DURATION:P5D
+SUMMARY:This is a task that is expected to take five days to complete
+STATUS:NEEDS-ACTION
+END:VTODO""")
+
+    # This assertion was failing before the fix
+    assert my_task.duration == timedelta(days=5)
+
+    # Verify the todo has the expected properties
+    assert my_task.get("UID") == "taskwithoutdtstart"
+    assert (
+        my_task.get("SUMMARY")
+        == "This is a task that is expected to take five days to complete"
+    )
+    assert "DURATION" in my_task
+    assert "DTSTART" not in my_task  # Confirming no DTSTART
+
+
+def test_todo_duration_preserves_property_access():
+    """Test that direct property access still works as expected."""
+    todo = Todo()
+    todo.add("UID", "test-property-access")
+    todo.add("DURATION", timedelta(hours=2))
+
+    # Direct property access should still work
+    assert todo["DURATION"].dt == timedelta(hours=2)
+
+    # And our computed property should match
+    assert todo.duration == timedelta(hours=2)
+
+
+# Event tests - same fix applies for consistency
+def test_event_duration_prefers_duration_property():
+    """Test that Event.duration also prefers `DURATION` property over calculated duration."""
+    event = Event()
```

Comment:
Could you check out if this works here: `@pytest.mark.parametrize("component_class", [Event, Todo])`?

This way, we can avoid code duplication.